src/dex/paraswap-limit-orders/paraswap-limit-orders.ts

```diff
@@ -0,0 +1,417 @@
+} from '../../constants';
+import { getBigIntPow, getDexKeysWithNetwork, wrapETH } from '../../utils';
+import { IDex } from '../../dex/idex';
+import { IDexHelper } from '../../dex-helper/idex-helper';
+import {
+  ParaswapLimitOrdersData,
+  ParaswapLimitOrderResponse,
+  ParaswapLimitOrderPriceSummary,
+  ParaswapPriceSummaryResponse,
+  OrderInfo,
+} from './types';
+import { ParaswapLimitOrdersConfig, Adapters } from './config';
+import { LimitOrderExchange } from '../limit-order-exchange';
+import { BI_MAX_UINT } from '../../bigint-constants';
+import augustusRFQABI from '../../abi/paraswap-limit-orders/AugustusRFQ.abi.json';
+import { ONE_ORDER_GASCOST } from './constant';
+
+export class ParaswapLimitOrders
+  extends LimitOrderExchange<
+    ParaswapLimitOrderResponse,
+    ParaswapPriceSummaryResponse
+  >
+  implements IDex<ParaswapLimitOrdersData>
+{
+  readonly hasConstantPriceLargeAmounts = false;
+  readonly needWrapNative = true;
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(ParaswapLimitOrdersConfig);
+
+  logger: Logger;
+
+  constructor(
+    protected network: Network,
+    protected dexKey: string,
+    protected dexHelper: IDexHelper,
+    protected adapters = Adapters[network] ? Adapters[network] : {},
+    protected augustusRFQAddress = ParaswapLimitOrdersConfig[dexKey][
+      network
+    ].rfqAddress.toLowerCase(),
+    protected rfqIface = new Interface(augustusRFQABI),
+  ) {
+    super(dexHelper.augustusAddress, dexHelper.provider);
+    this.logger = dexHelper.getLogger(dexKey);
+  }
+
+  get limitOrderProviderName() {
+    return LIMIT_ORDER_PROVIDERS.PARASWAP;
+  }
+
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return this.adapters[side] ? this.adapters[side] : null;
+  }
+
+  getIdentifier(srcToken: Address, destToken: Address) {
+    // Expected lowered Addresses
+    return `${this.dexKey.toLowerCase()}_${srcToken}_${destToken}`;
+  }
+
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    if (side === SwapSide.BUY) return [];
+
+    const _srcToken = wrapETH(srcToken, this.network);
+    const _destToken = wrapETH(destToken, this.network);
+
+    const _srcAddress = _srcToken.address.toLowerCase();
+    const _destAddress = _destToken.address.toLowerCase();
+
+    if (_srcAddress === _destAddress) {
+      return [];
+    }
+
+    const priceSummary = await this._getLatestPriceSummary(
+      _srcAddress,
+      _destAddress,
+    );
+    if (priceSummary === null) return [];
+
+    return [this.getIdentifier(_srcAddress, _destAddress)];
+  }
+
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<ParaswapLimitOrdersData>> {
+    if (side === SwapSide.BUY) return null;
```

Comment:
We should be supporting `BUY` for this.

Reply:
It was moved outside of current cycle as separate task after going prod:
https://linear.app/paraswap/issue/BACK-595/add-buy-suport-for-dex-limitorders

@erez-paraswap , is it ok or better to add to this cycle? It will involve changes on LO Service side too

Reply:
Let's leave it out of scope now and ship the sell side first. We'll address right after we ship and by then, LO will be out too